data_steward/cdr_cleaner/clean_cdr.py

```diff
@@ -200,498 +186,105 @@
-        LOGGER.info(
-            f"Dataset is unspecified.  Using default value of:\t{dataset_id}")
-
-    sandbox_dataset_id = sandbox.create_sandbox_dataset(project_id=project_id,
-                                                        dataset_id=dataset_id)
-
-    query_list = _gather_rdr_queries(project_id, dataset_id, sandbox_dataset_id)
-
-    LOGGER.info("Cleaning rdr_dataset")
-    clean_engine.clean_dataset(project_id, query_list, stage.RDR)
-
-
-def clean_ehr_dataset(project_id=None, dataset_id=None):
-    """
-    Run all clean rules defined for the ehr dataset.
-
-    :param project_id:  Name of the BigQuery project.
-    :param dataset_id:  Name of the dataset to clean
-    """
-    if project_id is None:
-        project_id = app_identity.get_application_id()
-        LOGGER.info(
-            f"Project is unspecified.  Using default value of:\t{project_id}")
-
-    if dataset_id is None:
-        dataset_id = bq_utils.get_dataset_id()
-        LOGGER.info(
-            f"Dataset is unspecified.  Using default value of:\t{dataset_id}")
-
-    sandbox_dataset_id = sandbox.create_sandbox_dataset(project_id=project_id,
-                                                        dataset_id=dataset_id)
-
-    query_list = _gather_ehr_queries(project_id, dataset_id, sandbox_dataset_id)
-
-    LOGGER.info("Cleaning ehr_dataset")
-    clean_engine.clean_dataset(project_id, query_list, stage.EHR)
-
-
-def clean_unioned_ehr_dataset(project_id=None, dataset_id=None):
-    """
-    Run all clean rules defined for the unioned ehr dataset.
-
-    :param project_id:  Name of the BigQuery project.
-    :param dataset_id:  Name of the dataset to clean
-    """
-    if project_id is None:
-        project_id = app_identity.get_application_id()
-        LOGGER.info(
-            f"Project is unspecified.  Using default value of:\t{project_id}")
-
-    if dataset_id is None:
-        dataset_id = bq_utils.get_unioned_dataset_id()
-        LOGGER.info(
-            f"Dataset is unspecified.  Using default value of:\t{dataset_id}")
-
-    sandbox_dataset_id = sandbox.create_sandbox_dataset(project_id=project_id,
-                                                        dataset_id=dataset_id)
-
-    query_list = _gather_unioned_ehr_queries(project_id, dataset_id,
-                                             sandbox_dataset_id)
-
-    LOGGER.info("Cleaning unioned_dataset")
-    clean_engine.clean_dataset(project_id, query_list, stage.UNIONED)
+    engine_parser = args_parser.get_base_arg_parser()
+    engine_parser.add_argument(
+        '-a',
+        '--data_stage',
+        required=True,
+        dest='data_stage',
+        action='store',
+        type=DataStage,
+        choices=list([s for s in DataStage if s is not DataStage.UNSPECIFIED]),
+        help='Specify the dataset')
+    return engine_parser
 
 
-def clean_combined_dataset(project_id=None, dataset_id=None):
+def get_dynamic_parser(optional_args):
```

Comment:
I'm not sure that/why a parser is needed after the arguments have already been consumed- let's discuss this soon. I think a dynamic parser might be viable in the long-run but for now just turning input into kwargs should be sufficient.

Reply:
Yes, the `dynamic_parser` is one way to achieve that. It dynamically creates parser args based on the input.